### PR TITLE
CLOUDP-124556: Improve error message

### PR DESCRIPTION
**Pull request by blva**
## Proposed changes

_Jira ticket:_ [CLOUDP-124556](https://jira.mongodb.org/browse/CLOUDP-124556)

Closes #[issue number]

## Checklist

- [ ] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [ ] I have updated [e2e/E2E-TESTS.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/e2e/E2E-TESTS.md) (if an e2e test has been added)
- [ ] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/cli/atlas/quickstart/quick_start.go [hunk 1]
```diff
@@ -130,6 +130,12 @@ func (opts *Opts) initStore(ctx context.Context) func() error {
 	}
 }
 
+func (opts *Opts) quickstartPreRun() error {
+	return opts.PreRunE(
```
Comment (blva) [inline internal/cli/atlas/quickstart/quick_start.go:134]: move validation to quickstart only

File: internal/cli/atlas/quickstart/quick_start.go [hunk 2]
```diff
@@ -130,6 +130,12 @@ func (opts *Opts) initStore(ctx context.Context) func() error {
...
+		opts.ValidateProjectID,
+	)
+}
+
 func (opts *Opts) PreRun(ctx context.Context, outWriter io.Writer) error {
 	opts.setTier()
```
Comment (andreaangiolillo) [inline internal/cli/atlas/quickstart/quick_start.go:140]: [q] Why didn’t you add this validation in the existing `PreRun`? we are already validating `CurrentIP` there

**Review by andreaangiolillo — COMMENTED** [review]
Only one question

File: internal/cli/atlas/quickstart/quick_start.go (hunk 2 above)
Comment (blva) [reply internal/cli/atlas/quickstart/quick_start.go:140]: The`ValidateProjectID` is the validation that throws `required flag(s) "%s" not set`, which means it's quickstart specific (other commands that use quickstart Run and PreRun manage their own flags). To make sure this validation is done only at the quickstart command itself we decouple it from `PreRun` which is publicly used by `setup`.

Comment (andreaangiolillo) [reply internal/cli/atlas/quickstart/quick_start.go:140]: ah, I see that setup is using `quickstart.PreRun` indeed. I am wondering if stop using the quickstart.PreRun would be a better way but this is no blocking

**Review by andreaangiolillo — APPROVED** [review]

File: internal/cli/atlas/quickstart/quick_start.go (hunk 2 above)
Comment (blva) [reply internal/cli/atlas/quickstart/quick_start.go:140]: Thnaks! The idea is to keep `PreRun` with as many validations that can be shared as possible, and keep the individual ones in a private method.

**Pull request merged**